pkg/build/builder/common.go

```diff
@@ -81,3 +84,47 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// runScriptInContainer runs a Bash script in an ephemeral Docker container
+// started off imageID, and returns an error if the script cannot be run or
+// returns a non-zero exit code.
+func runScriptInContainer(client DockerClient, script, imageID string) error {
+	glog.Infof("Running post commit hook with image %s ...", imageID)
+
+	// Read CPU and memory limits from /sys/fs/cgroup.
+	// These limits are used to enforce limits on the container that runs
+	// the post commit hook.
+	r := intFromFileReader{}
+	cpuShares := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	cpuPeriod := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	cpuQuota := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	memory := r.read("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if r.err != nil {
```

Comment:
Then I suggest reading the blog post, and see how [`bufio.Scanner`](https://golang.org/pkg/bufio/#Scanner) works ;)

The alternative would be lengthier and filled with a lot of code duplication.

The pattern I'm using here is meant to address this kind of smell:

```go
_, err = fd.Write(p0[a:b])
if err != nil {
    return err
}
_, err = fd.Write(p1[c:d])
if err != nil {
    return err
}
_, err = fd.Write(p2[e:f])
if err != nil {
    return err
}
// and so on
```